Tests for Category class

Gal-f opened the pull request:
1. Rearranging the models methods and creating a view method for categories to be used with tests
2. Adding automated tests for Category class

Resolves #45

Gal-f committed:
Rearranging methods in class Category

Moving a method from class Message to class Category
and adding a method for getting all the categories

Gal-f committed:
Category tests

Adding tests for methods in the Category class

simzacks commented on forumMessages/models.py:
```diff
@@ -6,6 +6,16 @@
     # Category.id field auto-generated by Django, will not be used
+    
+    @classmethod
+    def addCategory(self, categoryName):
```
The standard is that the first parameter in a classmethod is cls, not self.
This is because you are accessing the class and not an instance of it.

simzacks commented on forumMessages/models.py:
```diff
@@ -6,6 +6,16 @@
+    
+    @classmethod
+    def addCategory(self, categoryName):
+        self.categories.add(Category.objects.filter(name=categoryName))
```
what does this do?
There is no categories attribute in the Category class

Gal-f replied:
The method addCategory is under Message class, not Category class.
What it does (supposedly, at least...) is add  a Category object to the categories attribute of the Message instance (self).

Gal-f replied:
I see the problem. This got the other way around, probably committed in the wrong order and the method jumped back to Category class. I'll fix it.

ravidbro commented on forumMessages/models.py:
```diff
@@ -6,6 +6,16 @@
+    
+    @classmethod
+    def addCategory(self, categoryName):
+        self.categories.add(Category.objects.filter(name=categoryName))
```
1. in @classmethod you usually use 'cls' and not 'self' 
2. 'self' is referring to the class you are currently in which is Category and doesn't have a field 'categories'

Gal-f committed:
Adding a method for creating category

And moving the method for adding an existing category to a message
down to the Message class methods.

ravidbro commented on forumMessages/tests.py:
```diff
@@ -1,2 +1,21 @@
+    def test_getCategories(cls):
+        allCat = Category.getCategories()
+        assert isinstance(allCat, QuerySet)  # Check if the list of categories is OK after adding a test category
+        assert all(isinstance(cat, Category) for cat in Category)  # Check if all the instances in Category table
```
I'm missing assertion of the values inside the object,
what you asserted until here is pure Django behavior
